go-controller/pkg/util/subnet_annotations_unit_test.go

```diff
@@ -249,26 +252,25 @@ func TestCreateNodeHostSubnetAnnotation(t *testing.T) {
 	tests := []struct {
 		desc            string
 		inpDefSubnetIps []*net.IPNet
-		outExp          map[string]interface{}
+		outExp          map[string]string
 		errExp          bool
 	}{
 		{
 			desc:            "success path, valid default subnets",
 			inpDefSubnetIps: ovntest.MustParseIPNets("192.168.1.12/24"),
-			outExp: map[string]interface{}{
-				"k8s.ovn.org/node-subnets": "{\"default\":\"192.168.1.12/24\"}",
+			outExp: map[string]string{
+				"k8s.ovn.org/node-subnets": "{\"default\":[\"192.168.1.12/24\"]}",
 			},
 		},
 		{
-			desc: "success path, inpDefSubnetIps is nil",
-			outExp: map[string]interface{}{
-				"k8s.ovn.org/node-subnets": "{\"default\":[]}",
-			},
+			desc:   "success path, inpDefSubnetIps is nil",
+			outExp: map[string]string{},
```

Comment:
I don't quite understand your question. when there is hostsubnet available, we should not annotate this node at all, right?

Reply:
I mean that before this change, when we didn't have a `node-subnet` available we returned an empty list - i.e. `"k8s.ovn.org/node-subnets": "{\"default\":[]}"` .

Now, you've changed this behavior to return an empty map. Is this necessary ? Was this also required before but was missed ? I do not understand why you have changed this behavior. 

Just trying to understand - feels like I'm missing something.

Reply:
I see. Not sure if this can become a problem for other systems that consume this annotation.

At the very least, we need this referenced in the release notes.

Reply:
I agree